src/test/ui/type-check/assignment-in-if.rs

```diff
@@ -45,4 +45,9 @@ fn main() {
         //~| HELP did you mean to compare equality?
         println!("{}", x);
     }
+    if (if true { x = 4 } else { x = 5 }) {
+        //~^ ERROR mismatched types
+        //~| HELP did you mean to compare equality?
```

Comment:
Nit: we are not (currently) checking these annotations in ui tests (I want to...), but it doesn't appear that we emit a HELP in this case?